tests/mocks.py

```diff
@@ -9,6 +10,17 @@ class MockAzureCredential(AsyncTokenCredential):
     async def get_token(self, uri):
         return MockToken("", 9999999999, "")
 
+    async def close(self) -> None:
```

Comment:
I'm still perplexed as to why we didn't need this in azure-search-openai-demo, but maybe we have different mypy settings.

Reply:
I might be wrong but I think that sometimes when there are too much files/code Mypy doesn't show all of the errors 🤔 
I reviewed the setup there and it's the Same as ours except for the python version.